core/trino-main/src/main/java/io/trino/simd/BlockEncodingSimdSupport.java

```diff
@@ -110,6 +125,37 @@ enum X86SimdInstructionSet {
         }
     }
 
+    private static SimdSupport detectArmSimdSupport()
+    {
+        enum ArmSimdInstructionSet {
+            sve,
+        }
+
+        Set<String> flags = readCpuFlags();
+        EnumSet<ArmSimdInstructionSet> armFlags = EnumSet.noneOf(ArmSimdInstructionSet.class);
+
+        if (!flags.isEmpty()) {
+            for (ArmSimdInstructionSet instructionSet : ArmSimdInstructionSet.values()) {
+                if (flags.contains(instructionSet.name())) {
+                    armFlags.add(instructionSet);
+                }
+            }
+        }
+
+        // Ensure the JVM reports a sufficiently wide preferred vector size before enabling Vector API paths
+        if (PREFERRED_BIT_WIDTH < MINIMUM_SIMD_LENGTH) {
+            return SimdSupport.NONE;
+        }
+        else {
+            boolean hasSve = armFlags.contains(ArmSimdInstructionSet.sve);
```

Comment:
Maybe just inline this into the `PREFERRED_BIT_WIDTH` branch, and return `SimdSupport.NONE`